rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,200 @@
+
+### **Base font size**
+After auditing the whole product it came out that we are mostly using `13px` & `14px` as a base font-size on the body text. For the internal products, it is better to go with **`14px`** as a base font-size. Because for dashboards the content is pretty dense and we need to maintain a small font-size ratio compared to marketing/landing pages. This will help the users to consume more content into a readable format. Although, we can still play with spacings so that it doesn't feel much complex.
+
+### **Line Height**
+Line heights also plays a significant role in the whole typography spectrum. If the font-size is smaller the text will be cramped and difficult to read, instead if it is larger the eyes can get lost. So for optimal readability & accessibility aim for about `140% - 180%` line-height.
+
+Considering blade we generally have a simple formula that derives the line-height and we consider it to be 150% of the font-size. i.e,
+
+>**`line-height = font-size * 1.5`**
+
+And will round-off the resulting value.
+
+### **Current Limitations**
+There are two major limitations while scaling typography to multiple breakpoints,
+1. **It does not cater to landing/marketing pages.** Which are generally much larger compared to the size-ratio on the dashboard (Maybe a [Major Third](https://type-scale.com/?size=14&scale=1.125&text=A%20Visual%20Type%20Scale&font=Lato&fontweight=400&bodyfont=Poppins&bodyfontweight=400&lineheight=1.65&backgroundcolor=white&fontcolor=%23333&preview=false) scale will work better). This is currently being looked at by the comm. design team based on the same principles.
+2. **Same type scaling across all devices/breakpoints.** After the type scale is defined, we also need to define the scale across multiple device sizes. There are few approaches that we've came up with and after research from other DS/products like *GitHub*, *Eightshapes*, *Paystack*, *Stripe*, and a few more we've observed that for dashboards generally tweaking the **heading** sizes works best and the base font-size will remain same for all other devices/breakpoints.
+
+We'll look into the responsive type scale and see what approach fits best for our use case.
+
+>📝 Note: **Responsive Typography** will scale in segments. Which means if `h1` at `1440px` is `40px`, then at `991px` it will become `32px` directly rather than going from `40px -> 38px -> 36px -> 34px -> 32px...`.
+
+### **Our Approach**
```

Comment:
Can we first talk about our general approach and examples for body, paragraphs and everything and then go to headings? Because right now it implies that we are talking about only headings which we aren't actually.  

Something similar to [this](https://spectrum.adobe.com/page/typography/#Font-sizes)